### ISSUE TITLE
IFC Beams generated in any Elements version post 2.15 will not import into Navisworks, Revit or Rhino

### ISSUE THREAD

**Issue by secretlyagoblin** [hypar-io/Elements#189]
**Describe the bug**
Run in Hypar version 2.15 this creates a faceted IFC extrusion that can be read into Revit, Navisworks and Rhino (Rhino via Geomtry Gym)

Run in any version post 2.15, IFC import fails silently in Navis and Revit, in Rhino the following error is recorded:

> XXX Representation item IfcSurfaceCurveSweptAreaSolid not generated in #68= IFCBEAM('0ABW2RZcj0sOdWv60q3jnw',$,$,$,$,#67,#66,$);
XXX Unrecognized product representation #66= IFCPRODUCTDEFINITIONSHAPE($,$,(#65));

**To Reproduce**
```
            var model = new Model();

            var profile = new Profile(Polygon.Circle());
            var framingType = new StructuralFramingType(profile.Name, profile, 
                     BuiltInMaterials.Steel);
            var line = new Line(Vector3.Origin, new Vector3(5, 0, 5));
            var linearBeam = new Beam(line, framingType);

            model.AddElement(linearBeam);
            model.ToIFC("‪test.ifc");
```
**Expected behavior**
2.15 behavior is fine for my purposes - IFC object appears as open extrusion but it's enough to spaceproof for now.

Ideally a closed extrusion should be generated.

**Screenshots**
2.15 Behaviour below:
![215 Behaviour](https://user-images.githubusercontent.com/13952064/65922742-ce340180-e442-11e9-96e6-ca4458eb5602.PNG)

**Desktop:**
 - OS: Windows 10

**Comment by ikeough** [comment]
The story here is that we started using `IfcSurfaceCurveSweptAreaSolid` so we could use it for all beam curve types only to find out that that type [isn't supported](https://forums.buildingsmart.org/t/ifcsurfacecurvesweptareasolid-does-not-consistently-orient-the-swept-profile/1533) in any of the IFC2X3 MVDs. 

@secretlyagoblin I'm considering updating Elements IFC to IFC4, to take advantage of the Design Transfer View MVD. Do the applications shown above all have IFC4 support?

**Comment by secretlyagoblin** [comment]
Hey Ian, thanks for looking into this - from the looks of things...

- [Navis lists IFC4 as a supported format](https://knowledge.autodesk.com/support/navisworks-products/learn-explore/caas/sfdcarticles/sfdcarticles/Supported-file-formats-and-applications-for-Autodesk-Navisworks-2009-to-2012.html)
- [Revit lists IFC4 as a supported format](https://knowledge.autodesk.com/support/revit-products/learn-explore/caas/CloudHelp/cloudhelp/2019/ENU/Revit-DocumentPresent/files/GUID-6708CFD6-0AD7-461F-ADE8-6527423EC895-htm.html)
- [IFC4 is supported by Rhino via geometrygym](https://geometrygym.wordpress.com/downloads-windows/)

So it looks like we should be all good. Go ahead, and I'll let you know if we hit any issues.

**Comment by ikeough** [comment]
Revisiting this issue after far too long. Things have definitely improved since the first version that you used to test. Shown below is import into Solibri of HSS pipe sections. Notably missing is the "hole" in the pipe, because we aren't exporting it yet. But you do get properly closed full extrusions. 

There's a second problem in #229 which is that beams specifically don't show up in other applications, which I think is related to the use of `IfcSurfaceCurveSweptAreaSolid` which, although supported by the IFC4 spec, is not implemented in a number of applications. 

Would you be willing to run the test above with 0.6.2 version of Elements so I can narrow down this issue?

![image](https://user-images.githubusercontent.com/1139788/79516770-2efee200-8001-11ea-94ac-97c5784d2678.png)

**Comment by ikeough** [comment]
We believe this is fixed with the work in #297, but please test with >0.7.x and let us know.
